src/sagemaker/jumpstart/types.py

```diff
@@ -1253,18 +1115,1090 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
+        ] = model_specs.default_payloads
+        self.hosting_resource_requirements: Optional[
+            Dict[str, int]
+        ] = model_specs.hosting_resource_requirements
+        self.hosting_instance_type_variants: Optional[
+            str
+        ] = model_specs.hosting_instance_type_variants
+        self.min_studio_sdk_version: Optional[str] = studio_specs.get("minServerVersion")
+        notebook_location_keys = studio_specs.get("notebookLocations")
+        if notebook_location_keys:
+            self.notebook_location_uris: Optional[Dict[str, str]] = (
+                s3_path_join("s3://", content_bucket, notebook_location_keys.get("demoNotebook"))
+                if notebook_location_keys.get("demoNotebook")
+                else None
+            )
+        self.model_provider_icon_uri: Optional[str] = None  # TODO: Missing in specs?
+        self.task: Optional[str] = studio_specs.get("problemType")
+        self.framework: Optional[str] = studio_specs.get("framework")
+        self.datatype: Optional[str] = studio_specs.get("dataType")
+        self.license: Optional[str] = studio_specs.get("license")
+        self.contextual_help: Optional[str] = studio_specs.get("contextualHelp")
+        self.model_dir: Optional[str] = None
+        # Deploy kwargs
+        self.model_data_download_timeout: Optional[str] = model_specs.deploy_kwargs.get(
+            "model_data_download_timeout"
+        )
+        self.container_startup_health_check_timeout: Optional[str] = model_specs.deploy_kwargs.get(
+            "container_startup_health_check_timeout"
+        )
+
+        if self.training_supported:
+            if model_specs.training_model_package_artifact_uris:
+                self.training_model_package_artifact_uri: Optional[
+                    str
+                ] = model_specs.training_model_package_artifact_uris.get(self._region)
+            self.training_artifact_compression_type: Optional[str] = studio_specs.get(
+                "trainingArtifactCompressionType"
+            )
+            self.training_artifact_s3_data_type: Optional[str] = studio_specs.get(
+                "trainingArtifactS3DataType"
+            )
+            self.hyperparameters: List[JumpStartHyperparameter] = model_specs.hyperparameters
+            self.training_script_uri: Optional[str] = (
+                s3_path_join("s3://", content_bucket, model_specs.training_script_key)
+                if model_specs.training_script_key is not None
+                else None
+            )
+            self.training_prepacked_script_uri: Optional[str] = (
+                s3_path_join("s3://", content_bucket, model_specs.training_prepacked_script_key)
+                if model_specs.training_prepacked_script_key is not None
+                else None
+            )
+            self.training_prepacked_script_version: Optional[
+                str
+            ] = None  # TODO: studio_specs.minTrainingScriptVersion?
```

Comment:
**value**: https://github.com/aws/sagemaker-model-hub/blob/7c79b7f09295cd4a5e7875ca4e275544e2f50e0f/src/sagemakermodelhub/content_generation/metadata/sdk/sdk_metadata.py#L569

**example**: https://github.com/aws/sagemaker-model-hub/blob/7c79b7f09295cd4a5e7875ca4e275544e2f50e0f/src/sagemakermodelhub/content_generation/metadata/generated_metadata/sdk/specs/community_models/pytorch-ic-vgg16/specs_v3.0.0.json#L94

Reply:
Ok - the field is missing from ModelSpecs so we would need that.